wgpu/gui/base.py

```diff
@@ -175,3 +175,72 @@ def _request_draw(self):
         the FPS is limited to avoid draining CPU and power.
         """
         raise NotImplementedError()
+
+
+class WgpuAutoGui:
+    """Mixin class for canvases implementing autogui."""
+
+    def handle_event(self, event):
+        """Handle an incoming event.
+
+        Subclasses can overload this method. Events include widget
+        resize, mouse/touch interaction, key events, and more. An event
+        is a dict with at least the key event_type. For details, see
+        https://jupyter-rfb.readthedocs.io/en/latest/events.html
+        """
+        event_type = event.get("event_type")
+        for callback in self._event_handlers[event_type]:
```

Comment:
Are you ok with assuming `self._event_handlers` exists?

Reply:
No, I'm not ;)